code/messaging/src/androidTest/java/com/adobe/marketing/mobile/MessagingFunctionalTestUtils.java

```diff
@@ -0,0 +1,277 @@
+     * to two map elements "xdm.stitchId" = "myID" and "xdm.eventType" = "myType".
+     * <p>
+     * Method is called recursively. To use, call with an empty path such as
+     * {@code addKeys("", new ObjectMapper().readTree(JsonNodeAsString), map);}
+     *
+     * @param currentPath the path in {@code JsonNode} to process
+     * @param jsonNode    {@link JsonNode} to deserialize
+     * @param map         {@code Map<String, String>} instance to store flattened JSON result
+     * @see <a href="https://stackoverflow.com/a/24150263">Stack Overflow post</a>
+     */
+    private static void addKeys(String currentPath, JsonNode jsonNode, Map<String, String> map) {
+        if (jsonNode.isObject()) {
+            ObjectNode objectNode = (ObjectNode) jsonNode;
+            Iterator<Map.Entry<String, JsonNode>> iter = objectNode.fields();
+            String pathPrefix = currentPath.isEmpty() ? "" : currentPath + ".";
+
+            while (iter.hasNext()) {
+                Map.Entry<String, JsonNode> entry = iter.next();
+                addKeys(pathPrefix + entry.getKey(), entry.getValue(), map);
+            }
+        } else if (jsonNode.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) jsonNode;
+
+            for (int i = 0; i < arrayNode.size(); i++) {
+                addKeys(currentPath + "[" + i + "]", arrayNode.get(i), map);
+            }
+        } else if (jsonNode.isValueNode()) {
+            ValueNode valueNode = (ValueNode) jsonNode;
+            map.put(currentPath, valueNode.asText());
+        }
+    }
+
+    static void dispatchEdgePersonalizationEventWithMessagePayload(final String name) {
+        final EventData eventData = new EventData();
+        final List<Variant> items = new ArrayList<>();
+        items.add(Variant.fromVariantMap(getVariantMapFromFile(name)));
+        eventData.putVariant("payload", Variant.fromVariantList(items));
+        final Event event = new Event.Builder("edge response testing", MessagingConstants.EventType.EDGE, MessagingConstants.EventSource.PERSONALIZATION_DECISIONS)
+                .setData(eventData)
+                .build();
+        MobileCore.dispatchEvent(event, new ExtensionErrorCallback<ExtensionError>() {
+            @Override
+            public void error(ExtensionError extensionError) {
+                Log.debug(LOG_TAG, "exception occurred: %s", extensionError.getErrorName());
+            }
+        });
+    }
+
+    static Map<String, Variant> getVariantMapFromFile(final String name) {
+        try {
+            final JSONObject json = new JSONObject(loadStringFromFile(name));
+            return toVariantMap(json);
+        } catch (final JSONException jsonException) {
+            Log.warning(LOG_TAG, "Exception occurred when creating the JSONObject: %s", jsonException.getMessage());
+            return null;
+        }
+    }
+
+    static Map<String, Object> getMapFromFile(final String name) {
+        try {
```

Comment:
Can we make same changes in this function too as per the comment on previous function and all other functions where they are applicable.